x-pack/plugins/observability_solution/synthetics/public/apps/synthetics/components/common/components/monitor_type_badge.tsx

```diff
@@ -5,74 +5,76 @@
  * 2.0.
  */
 
-import React, { MouseEvent, KeyboardEvent } from 'react';
-import { EuiBadge, EuiIcon } from '@elastic/eui';
-import { euiStyled } from '@kbn/kibana-react-plugin/common';
-import {
-  EncryptedSyntheticsMonitor,
-  ConfigKey,
-  FormMonitorType,
-  MonitorTypeEnum,
-} from '../../../../../../common/runtime_types';
+import React, { MouseEvent } from 'react';
+import { EuiBadge } from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+import { FormMonitorType, MonitorTypeEnum } from '../../../../../../common/runtime_types';
 
 export function MonitorTypeBadge({
-  monitor,
+  monitorType,
   ariaLabel,
   onClick,
-  onKeyPress,
 }: {
-  monitor: EncryptedSyntheticsMonitor;
+  monitorType: string;
   ariaLabel?: string;
-  onClick?: (evt: MouseEvent<HTMLDivElement>) => void;
-  onKeyPress?: (evt: KeyboardEvent<HTMLDivElement>) => void;
+  onClick?: () => void;
 }) {
-  const badge = (
-    <EuiBadgeStyled data-is-clickable={!!onClick}>
-      <EuiIcon size="s" type={getMonitorTypeBadgeIcon(monitor)} />{' '}
-      {getMonitorTypeBadgeTitle(monitor)}
-    </EuiBadgeStyled>
-  );
-
   return onClick ? (
-    <div title={ariaLabel} aria-label={ariaLabel} onClick={onClick} onKeyPress={onKeyPress}>
-      {badge}
-    </div>
+    <EuiBadge
+      onClick={onClick}
+      onClickAriaLabel={getFilterTitle(monitorType)}
+      title={ariaLabel}
+      aria-label={ariaLabel}
+      iconType={getMonitorTypeBadgeIcon(monitorType)}
+      onMouseDown={(e: MouseEvent) => {
+        // Prevents the click event from being propagated to the @elastic/chart metric
+        e.stopPropagation();
+      }}
+    >
+      {getMonitorTypeBadgeTitle(monitorType)}
+    </EuiBadge>
   ) : (
-    badge
+    <EuiBadge
+      title={ariaLabel}
+      aria-label={ariaLabel}
+      iconType={getMonitorTypeBadgeIcon(monitorType)}
+      onMouseDown={(e: MouseEvent) => {
+        // Prevents the click event from being propagated to the @elastic/chart metric
+        e.stopPropagation();
+      }}
+    >
+      {getMonitorTypeBadgeTitle(monitorType)}
+    </EuiBadge>
```

Comment:
You don't need to check if `onClick` is defined and have two variants for this component. If `onClick`  is undefined it'll pass undefined to the `EuiBadge` component.

Reply:
i tried it doesn't work with onClickAriaLabel being present always. it's kinda weird.